pkg/microservice/user/core/service/user/user.go

```diff
@@ -56,13 +59,62 @@ type SyncUserInfo struct {
 	Name         string `json:"name"`
 }
 
+func SearchAndSyncUser(si *systemconfig.Connector, logger *zap.SugaredLogger) error {
+	if si == nil || si.Config == nil {
+		logger.Error("can't find connector")
+		return fmt.Errorf("can't find connector")
+	}
+	config, ok := si.Config.(*ldap.Config)
+	if !ok {
+		return fmt.Errorf("connector config error")
+	}
+	l, err := ldapv3.Dial("tcp", config.Host)
+	if err != nil {
+		logger.Errorf("ldap dial host:%s error, error msg:%s", config.Host, err)
+		return err
+	}
+	defer l.Close()
+
+	err = l.Bind(config.BindDN, config.BindPW)
+	if err != nil {
+		logger.Errorf("ldap bind host:%s error, error msg:%s", config.Host, err)
+		return err
+	}
+
+	searchRequest := ldapv3.NewSearchRequest(
+		config.GroupSearch.BaseDN,
+		ldapv3.ScopeWholeSubtree, ldapv3.NeverDerefAliases, 0, 0, false,
+		config.GroupSearch.Filter,            // The filter to apply
+		[]string{config.UserSearch.NameAttr}, // A list attributes to retrieve
+		nil,
+	)
+
+	sr, err := l.Search(searchRequest)
```

Comment:
It will still cause search limit problem. With SearchWithPaging function it will bypass the soft limit by the server, but it cannot bypass the hard limit set by the server

Reply:
so, how to resolve it?